### ISSUE TITLE
tf->caffe error Tensorflow has not supported operator [Softmax] with name [eval_prediction]

### ISSUE THREAD

**Issue by jq460494839** [Microsoft/MMdnn#584]
Platform (ubuntu 16.04):

Python version:3.6

Source framework with version (tensorflow-gpu            1.12.0):

Destination framework with version (caffe-gpu                 1.0 ):

(py3-mmdnn) root@ubuntu:/home/j00465446/test/ckpt_model# mmconvert -sf tensorflow -in my_model.ckpt.meta -iw  my_model.ckpt --dstNode eval_prediction -df caffe -om outmodel
Parse file [my_model.ckpt.meta] with binary format successfully.
Tensorflow model file [my_model.ckpt.meta] loaded successfully.
Tensorflow checkpoint file [my_model.ckpt] loaded successfully. [27] variables loaded.
**Tensorflow has not supported operator [Softmax] with name [eval_prediction].**
IR network structure is saved as [2d6f6575a1414bd1a89c5db0b01b93b8.json].
IR network structure is saved as [2d6f6575a1414bd1a89c5db0b01b93b8.pb].
IR weights are saved as [2d6f6575a1414bd1a89c5db0b01b93b8.npy].
Parse file [2d6f6575a1414bd1a89c5db0b01b93b8.pb] with binary format successfully.
Target network code snippet is saved as [2d6f6575a1414bd1a89c5db0b01b93b8.py].
Target weights are saved as [2d6f6575a1414bd1a89c5db0b01b93b8.npy].
**WARNING: Logging before InitGoogleLogging() is written to STDERR
E0213 15:17:43.837237 49862 common.cpp:114] Cannot create Cublas handle. Cublas won't be available.
E0213 15:17:43.837512 49862 common.cpp:121] Cannot create Curand generator. Curand won't be** available.
I0213 15:17:43.837596 49862 net.cpp:51] Initializing net from parameters:
state {
  phase: TRAIN
  level: 0
}
layer {
  name: "eval_data"
  type: "Input"
  top: "eval_data"
  input_param {
    shape {
      dim: 1
      dim: 4
      dim: 1000
      dim: 1
    }
  }
}
layer {
  name: "Conv2D"
  type: "Convolution"
  bottom: "eval_data"
  top: "Conv2D"
  convolution_param {
    num_output: 16
    bias_term: true
    group: 1
    stride: 1
    pad_h: 0
    pad_w: 0
    kernel_h: 5
    kernel_w: 1
  }
}
layer {
  name: "Relu"
  type: "ReLU"
  bottom: "Conv2D"
  top: "Conv2D"
}
layer {
  name: "MaxPool"
  type: "Pooling"
  bottom: "Conv2D"
  top: "MaxPool"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
    pad_h: 0
    pad_w: 0
  }
}
layer {
  name: "Conv2D_1"
  type: "Convolution"
  bottom: "MaxPool"
  top: "Conv2D_1"
  convolution_param {
    num_output: 16
    bias_term: true
    group: 1
    stride: 1
    pad_h: 0
    pad_w: 0
    kernel_h: 5
    kernel_w: 1
  }
}
layer {
  name: "Relu_1"
  type: "ReLU"
  bottom: "Conv2D_1"
  top: "Conv2D_1"
}
layer {
  name: "MaxPool_1"
  type: "Pooling"
  bottom: "Conv2D_1"
  top: "MaxPool_1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
    pad_h: 0
    pad_w: 0
  }
}
layer {
  name: "fc_feature"
  type: "Flatten"
  bottom: "MaxPool_1"
  top: "fc_feature"
}
layer {
  name: "MatMul"
  type: "InnerProduct"
  bottom: "fc_feature"
  top: "MatMul"
  inner_product_param {
    num_output: 64
    bias_term: true
  }
}
layer {
  name: "Relu_2"
  type: "ReLU"
  bottom: "MatMul"
  top: "MatMul"
}
layer {
  name: "MatMul_1"
  type: "InnerProduct"
  bottom: "MatMul"
  top: "MatMul_1"
  inner_product_param {
    num_output: 2
    bias_term: true
  }
}
I0213 15:17:43.837741 49862 layer_factory.hpp:77] Creating layer eval_data
I0213 15:17:43.837772 49862 net.cpp:84] Creating Layer eval_data
I0213 15:17:43.837781 49862 net.cpp:380] eval_data -> eval_data
I0213 15:17:43.837941 49862 net.cpp:122] Setting up eval_data
I0213 15:17:43.837960 49862 net.cpp:129] Top shape: 1 4 1000 1 (4000)
I0213 15:17:43.837965 49862 net.cpp:137] Memory required for data: 16000
I0213 15:17:43.837976 49862 layer_factory.hpp:77] Creating layer Conv2D
I0213 15:17:43.837991 49862 net.cpp:84] Creating Layer Conv2D
I0213 15:17:43.837997 49862 net.cpp:406] Conv2D <- eval_data
I0213 15:17:43.838013 49862 net.cpp:380] Conv2D -> Conv2D
**F0213 15:17:43.838270 49862 cudnn_conv_layer.cpp:52] Check failed: error == cudaSuccess (35 vs. 0)  CUDA driver version is insufficient for CUDA runtime version**
*** Check failure stack trace: ***
Aborted (core dumped)

**Comment by rainLiuplus** [comment]
Hi @jq460494839 , MMdnn only supports convert the network architecture. Please specify dstNodeName as the end point of the network architecture.

**Comment by jq460494839** [comment]
Tensorflow has not supported operator [Softmax] with name [eval_prediction].
I don't know where the mistake is.
The 'eval_prediction' is the end point

**Comment by jq460494839** [comment]
> Hi @jq460494839 , MMdnn only supports convert the network architecture. Please specify dstNodeName as the end point of the network architecture.

Tensorflow has not supported operator [Softmax] with name [eval_prediction].
I don't know where the mistake is.
The 'eval_prediction' is the end point

**Comment by rainLiuplus** [comment]
Hi @jq460494839 , the _eval_prediction_  looks like a node in loss stage.  The output node type of the network is often fc, matmul, etc and in your case it is probably the input node of _eval_prediction_ . You can try it this way, thanks!

**Comment by jq460494839** [comment]
> Hi @jq460494839 , the _eval_prediction_ looks like a node in loss stage. The output node type of the network is often fc, matmul, etc and in your case it is probably the input node of _eval_prediction_ . You can try it this way, thanks!

Hi,@rainLiuplus
thank you for your help, i set dstNode with the lasted fc layer, and it works!
